samples/deployment/test-network/setup.sh

```diff
@@ -12,81 +12,67 @@ if [[ -z "${FPC_PATH}" ]]; then
-  else
-    cp "${FILE}" "${BACKUP}"
-  fi
-}
+
+##############################################################################################
+# Get the fabric samples repo and fetch binaries and container images
+##############################################################################################
 
 echo "Prepare fabric samples test-network for FPC"
 
 FABRIC_SAMPLES=${FPC_PATH}/samples/deployment/test-network/fabric-samples
-CORE_PATH=${FABRIC_SAMPLES}/config/core.yaml
-DOCKER_PATH=${FABRIC_SAMPLES}/test-network/docker
-DOCKER_COMPOSE_TEST_NET=${DOCKER_PATH}/docker-compose-test-net.yaml
-DOCKER_COMPOSE_CA=${DOCKER_PATH}/docker-compose-ca.yaml
 
-if [ ! -d "${FABRIC_SAMPLES}/bin" ]; then
-  echo "Error: no fabric binaries found, see README.md"
-  exit 1
+if [ ! -d "${FABRIC_SAMPLES}" ]; then
+  echo "Fabric samples not found! cloning now!"
+  git clone https://github.com/hyperledger/fabric-samples "${FABRIC_SAMPLES}"
+  pushd "${FABRIC_SAMPLES}"
+  git checkout -b "works" 50b69f6
+  popd
 fi
 
-
-###############################################
-# Adding FPC support to core.yaml
-###############################################
-
-echo "Adding FPC external builder to core.yaml"
-# Create a backup copy of `core.yaml` first and always work from there.
-backup ${CORE_PATH}
-yq eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' -i ${CORE_PATH} core_ext.yaml
+if [ ! -d "${FABRIC_SAMPLES}/bin" ]; then
+  echo "Error: no fabric binaries found. Let's run 'network.sh prereq'"
+  pushd "${FABRIC_SAMPLES}/test-network"
+  ./network.sh prereq
+  popd
+fi
 
 
-###############################################
+##############################################################################################
 # Resolve relative paths for docker volumes
-###############################################
+##############################################################################################
 
 # Also there is another issue with this approach here. When working completely inside the FPC dev-container,
 # the volume mounts won't work. The reason is that the docker daemon provided by the host cannot parse volume paths.
 # For instance, ${FPC_PATH} inside dev-container is `/project/src/github.com/hyperledger/fabric-private-chaincode/`.
 # The correct path from the docker daemon perspective is something like
 # `/Users/marcusbrandenburger/Developer/gocode/src/github.com/hyperledger/fabric-private-chaincode/` (in my case on the mac).
 # For this reason we use DOCKERD_FPC_PATH to resolve relative host paths.
-if [ ! -z ${DOCKERD_FPC_PATH+x} ]; then
+if [ -n "${DOCKERD_FPC_PATH+x}" ]; then
   echo "Oo we are in docker mode! we need to use the host fpc path"
   FPC_PATH_HOST=${DOCKERD_FPC_PATH}
 else
   FPC_PATH_HOST=${FPC_PATH}
 fi
 echo "set FPC_PATH_HOST = ${FPC_PATH_HOST}"
 
-FABRIC_SAMPLES_HOST=${FPC_PATH_HOST}/samples/deployment/test-network/fabric-samples
+FABRIC_SAMPLES_HOST=${FPC_PATH}/samples/deployment/test-network/fabric-samples
```

Comment:
Does this still take into account the docker/non-docker usage?

Reply:
ok, here you might be right, because I changed something for the docker situation to work but maybe the not docker one wouldn't ......